src/main/java/uk/co/real_logic/benchmarks/rtt/Configuration.java

```diff
@@ -0,0 +1,543 @@
+        this.iterations = checkMinValue(builder.iterations, 1, "Iterations");
+        this.numberOfMessages = checkMinValue(builder.numberOfMessages, 1, "Number of messages");
+        this.burstSize = checkMinValue(builder.burstSize, 1, "Burst size");
+        this.messageSize = checkMinValue(builder.messageSize, MIN_MESSAGE_SIZE, "Message size");
+        this.messageProviderClass = validateMessageProviderClass(builder.messageProviderClass);
+        this.senderIdleStrategy = requireNonNull(builder.senderIdleStrategy, "Sender IdleStrategy cannot be null");
+        this.receiverIdleStrategy = requireNonNull(builder.receiverIdleStrategy, "Receiver IdleStrategy cannot be null");
+    }
+
+    /**
+     * Number of the warm-up iterations, where each iteration has a duration of one second. Warm-up iterations results
+     * will be discarded.
+     *
+     * @return number of the warm-up iterations, defaults to {@link #DEFAULT_WARM_UP_ITERATIONS}.
+     */
+    public int warmUpIterations()
+    {
+        return warmUpIterations;
+    }
+
+    /**
+     * Number of messages that must be sent every iteration during warm-up.
+     *
+     * @return number of messages to be sent every warm-up iteration, defaults to
+     * {@link #DEFAULT_WARM_UP_NUMBER_OF_MESSAGES}.
+     * @implNote Actual number of messages sent can be less than this if the underlying implementation is not capable
+     * of achieving this rate.
+     */
+    public int warmUpNumberOfMessages()
+    {
+        return warmUpNumberOfMessages;
+    }
+
+    /**
+     * Number of the measurement iterations, where each iteration has a duration of one second.
+     *
+     * @return number of the measurement iterations, defaults to {@link #DEFAULT_ITERATIONS}.
+     */
+    public int iterations()
+    {
+        return iterations;
+    }
+
+    /**
+     * Number of messages that must be sent every iteration during measurement.
+     *
+     * @return number of messages to be sent every measurement iteration
+     * @implNote Actual number of messages sent can be less than this if the underlying implementation is not capable
+     * of achieving this rate.
+     */
+    public int numberOfMessages()
+    {
+        return numberOfMessages;
+    }
+
+    /**
+     * Number of messages to be sent at once. For example given {@link #warmUpNumberOfMessages()} of {@code 1000} and a
+     * {@link #burstSize()} of {code 1} there will be one message sent every millisecond. However if the
+     * {@link #burstSize()} is {@code 5} then there will be a batch of five messages sent every five milliseconds.
+     *
+     * @return number of messages to be sent at once, defaults to {@link #DEFAULT_BURST_SIZE}.
+     */
+    public int burstSize()
+    {
+        return burstSize;
+    }
+
+    /**
+     * Size of message in bytes.
+     *
+     * @return size of message in bytes, defaults to {@link #MIN_MESSAGE_SIZE}.
+     */
+    public int messageSize()
```

Comment:
In general we prefer to use the term "length" rather than size when referring to a single dimension, e.g. a two dimensional buffer has a size which is a product of length times width. In computing we are terrible with terms.

Reply:
How about `messageLength` and `burstSize` (or `batchSize`)?